src/app/dao/tests/FirebaseFirestoreDAOMock.spec.ts

```diff
@@ -171,11 +165,14 @@ export abstract class FirebaseFirestoreDAOMock<T extends FirebaseJSONObject> imp
 
             }
         }
-        return null;
+        return new Subscription(() => {
+            this.callbacks = this.callbacks.filter(
+                (value: [FirebaseCondition[], FirebaseCollectionObserver<T>]): boolean => {
+                    return value[0] !== conditions && value[1] !== callback;
```

Comment:
le !== et === sont trustworthy pour des FirebaseCondition ???

Reply:
ici oui, on veut voir l'égalité physique pour bien supprimer ceux qu'on ajoute dans la même fonction

Reply:
La subscription que tu retourne est une "unsubsciption" dont le code veut dire 
if value[0] == conditions && value[1] === callback => remove this callback from the list, right ?

Si oui, tu as mal négativé a && b, c'est non-a or non-b, si je ne m'abuse.

Reply:
vrai, j'ai mieux négativé maintenant en (a && b) === false pour être plus clair !